third_party/llvm_openmp/openmp.bzl

```diff
@@ -19,3 +30,72 @@ def dict_add(*dictionaries):
     for d in dictionaries:
         result.update(d)
     return result
+
+def select_os_specific(L, M, W):
+    return select({
+        "@org_tensorflow//tensorflow:linux_x86_64": L,
+        "@org_tensorflow//tensorflow:macos": M,
+        "@org_tensorflow//tensorflow:windows": W,
+        "//conditions:default": L,
+    })
+
+def libname_os_specific():
+    return "" + select_os_specific(L = "libiomp5.so", M = "libiomp5.dylib", W = "libiomp5md.dll")
+
+# TODO(Intel-tf) Replace the following calls to cc_binary with cc_library.
+# cc_library should be used for files that are not independently executed. Using
+# cc_library results in linking errors. For e.g on Linux, the build fails
+# with the following error message.
+# ERROR: //tensorflow/BUILD:689:1: Linking of rule '//tensorflow:libtensorflow_framework.so.2.4.0' failed (Exit 1)
+# /usr/bin/ld.gold: error: symbol GOMP_parallel_loop_nonmonotonic_guided has undefined version VERSION
+# /usr/bin/ld.gold: error: symbol GOMP_parallel_start has undefined version GOMP_1.0
+# /usr/bin/ld.gold: error: symbol GOMP_cancellation_point has undefined version GOMP_4.0
+# /usr/bin/ld.gold: error: symbol omp_set_num_threads has undefined version OMP_1.0
+# ......
+# ......
+
+# MacOS build has not been tested, however since the MacOS build of openmp
+# uses the same configuration as Linux, the following should work.
+def libiomp5_cc_binary(libname, cppsources, srcdeps, common_includes):
+    cc_binary(
+        name = libname,
+        srcs = cppsources + srcdeps +
+            select_os_specific(
```

Comment:
Please create another rule `if_linux_or_macos(T, F)` that returns `T` when it's linux or macos, then use this clause for everything that linux and macos use the same values.